src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,104 @@
-export const Pagination = () => {};
+import { MouseEvent } from 'react';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total, perPage, currentPage = 1, onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+  const isFirstPage = currentPage <= 1;
+  const isLastPage = currentPage >= totalPages;
+
+  const handlePageClick = (
+    e: MouseEvent<HTMLAnchorElement, globalThis.MouseEvent>,
+    id: number,
+  ) => {
+    e.preventDefault();
+
+    onPageChange(id);
+  };
+
+  const handleNextPage = (
+    e: MouseEvent<HTMLAnchorElement, globalThis.MouseEvent>,
+
+  ) => {
+    e.preventDefault();
+
+    if (isLastPage) {
+      return;
+    }
+
+    onPageChange(currentPage + 1);
+  };
+
+  const handlePrevPage = (
+    e: MouseEvent<HTMLAnchorElement, globalThis.MouseEvent>,
+  ) => {
+    e.preventDefault();
+
+    if (isFirstPage) {
+      return;
+    }
+
+    onPageChange(currentPage - 1);
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={cn('page-item', {
+        disabled: isFirstPage,
+      })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={isFirstPage}
+          onClick={handlePrevPage}
+        >
+          «
+        </a>
+      </li>
+      {new Array(totalPages).fill(null).map((_, idx) => {
```

Comment:
it's ok, yet you can make a little bit simplier
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#specifications
https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

I like this variant
```
[...Array(totalPages).keys()]
```

Reply:
I have ts error for your variant: 
`Type 'IterableIterator<number>' can only be iterated through when using the '--downlevelIteration' flag or with a '--target' of 'es2015' or higher.ts(2802)`

But I use other variant from link.